### ISSUE TITLE
Restart pagination after a specific page break

### ISSUE THREAD

**Issue by s2software** [dompdf/dompdf#1878]
I am using the following code to print pagination at the footer of each page:
```php
<script type="text/php">	// (activate dompdf option isPhpEnabled)
  if (isset($pdf))
  {
    $font = $fontMetrics->get_font("helvetica", "bold");
    $size = 8;
    $color = array(0,0,0);
    $word_space = 0.0;  //  default
    $char_space = 0.0;  //  default
    $angle = 0.0;   //  default
    //
    $text = "Page {PAGE_NUM} of {PAGE_COUNT}";
    $text_width = $fontMetrics->getTextWidth($text, $font, $size);
    //
    $x = 690 - $text_width;
    $y = 825;
    $pdf->page_text($x, $y, $text, $font, $size, $color, $word_space, $char_space, $angle);
  }
</script>
```
Is there any way to reset the page counter after a specific page break?

Example:
Page 1, Page 2
`<span style="page-break-after: always;"></span>`
Page 1, Page 2, Page 3

This is needed to print in the same PDF two different "documents".

Alternative... Any way to render separately the two html and then join in one single PDF?
Thanks

**Comment by pixagraphic** [comment]
Did you find a way to do this? I need the exact same thing: reset page counter after explicit page-break. I tried to add "counter-reset: page;" to the page-break css class but it is not working.

**Comment by bsweeney** [comment]
@s2software that's going to require a bit of work. Page text is set and then rendered on every page after the document is put together. You can't really manipulate the number using solely that method. You could maybe do something with page_script instead (basically wrap what you have in a page_script method call and use `$pdf->text()` instead of `$pdf->page_text()`). But you would need to know how many pages each document will be. It's not impossible, but not easy.

If you can render each document separately and then combine them I'd recommend doing that. A library like FPDF/FPDI or libmergepdf should make it easy enough to do.



@pixagraphic your issue is different since you're trying to do this in the HTML. Via CSS you would just need to have an element that's styled to reset the page count. For example:

```
<div style="page-break-before: always; counter-reset: page 1;"></div>
```

**Comment by ggedde** [comment]
Same Issue here.

@bsweeney Unfortunately, this did not work (it did not increment on the subsequent pages just stayed as 1)
`<div style="page-break-before: always; counter-reset: page 1;"></div>`

Instead I got it to work by changing how the `counter-increment: page;` is set.
Originally it is set on `body` but I think once you reset it, it will no longer work on `body`.
So I moved it to a custom `body footer`
_\* Keep in mind this is with the `<footer>` above the page content_ 
```
body footer {
   ...
   counter-increment: page;
}
```
This worked, but it also made the initial pages increment by 2 as there were two `counter-increment` set. So you need to remove the one from the body. Also on the page counter reset it will start at 1, but the footer reset will make it go to 2. So you need to change the reset to `counter-reset: page;` which will set it to 0, then the footer reset will increase it to 1.
```
body {
   ...
   counter-increment: none;
}
```
and 
```
<div style="page-break-before: always; counter-reset: page;"></div>
```
**Then it works!**

**Comment by bsweeney** [comment]
Yeah, you are correct. Setting counter-reset adds a new page counter context, which I didn't consider in my response. I have a better example of using counters...somewhere in here.
